docs/source/tutorials/e2e_flow.rst

```diff
@@ -25,222 +25,239 @@ from the community seemlessly with torchtune.
+There are 4 types of folders:
+
+1) **recipe_state**: Holds recipe_state.pt with the information necessary to restart the last intermediate epoch. For more information, please check our deep-dive :ref:`Checkpointing in torchtune <understand_checkpointer>`.;
+2) **logs**: Defined in your config in metric_logger;
+3) **epoch_{}**: Contains your new trained model weights plus all original files of the model, except the checkpoints, making it easy for you to choose an specific epoch to run inference on or push to a model hub;
 
-The final trained weights are merged with the original model and split across two checkpoint files
-similar to the source checkpoints from the HF Hub
-(see the :ref:`LoRA Tutorial <lora_finetune_label>` for more details).
-In fact the keys will be identical between these checkpoints.
-We also have a third checkpoint file which is much smaller in size
-and contains the learnt LoRA adapter weights. For this tutorial, we'll only use the model
-checkpoints and not the adapter weights.
+.. code-block:: text
 
-|
+    $ tree -a /tmp/torchtune/llama3_2_3B/lora_single_device
+    /tmp/torchtune/llama3_2_3B/lora_single_device
+    ├── epoch_0
+    │   ├── adapter_config.json
+    │   ├── adapter_model.pt
+    │   ├── adapter_model.safetensors
+    │   ├── config.json
+    │   ├── ft-model-00001-of-00002.safetensors
+    │   ├── ft-model-00002-of-00002.safetensors
+    │   ├── generation_config.json
+    │   ├── LICENSE.txt
+    │   ├── model.safetensors.index.json
+    │   ├── original
+    │   │   ├── orig_params.json
+    │   │   ├── params.json
+    │   │   └── tokenizer.model
+    │   ├── original_repo_id.json
+    │   ├── README.md
+    │   ├── special_tokens_map.json
+    │   ├── tokenizer_config.json
+    │   ├── tokenizer.json
+    │   └── USE_POLICY.md
+    ├── logs
+    │   └── log_1734652101.txt
+    └── recipe_state
+        └── recipe_state.pt
+
+Let's understand the files:
+
+- `adapter_model.safetensors` and `adapter_model.pt` are your LoRA trained adapter weights. We save a duplicated .pt version of it to facilitate resuming from checkpoint.
+- `ft-model-{}-of-{}.safetensors` are your trained full model weights (not adapters). When LoRA finetuning, these are only present if we set ``save_adapter_weights_only=False``. In that case, we merge the merged base model with trained adapters, making inference easier.
+- `adapter_config.json` is used by Huggingface PEFT when loading an adapter (more on that later);
+- `model.safetensors.index.json` is used by Huggingface .from_pretrained when loading the model weights (more on that later)
+- All other files were originally in the checkpoint_dir. They are automatically copied during training. Files over 100MiB and ending on .safetensors, .pth, .pt, .bin are ignored, making it lightweight.
```

Comment:
should we  add double quotations here for better rendering?

![image](https://github.com/user-attachments/assets/698f4a1e-ac24-4421-977d-4dd60dd69f53)